src/main/java/com/liferay/faces/util/cache/CacheFactory.java

```diff
@@ -0,0 +1,228 @@
+	 * FactoryExtensionFinder}. The returned instance is designed to be accessed and modified by multiple threads
+	 * concurrently, so it is guaranteed to be {@link java.io.Serializable}. The returned cache will avoid exceeding the
+	 * maximum cache capacity. The algorithm used to ensure that the cache size does not exceed the max capacity (for
+	 * example least-recently-used) may cause significant performance differences between the returned cache and a cache
+	 * returned from {@link #getConcurrentCache()} or {@link #getConcurrentCache(int)}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 * @param  maxCapacity      The maximum capacity of the cache.
+	 */
+	public static <K, V> Cache<K, V> getConcurrentCacheInstance(ExternalContext externalContext, int initialCapacity,
+		int maxCapacity) {
+
+		CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getConcurrentCache(initialCapacity, maxCapacity);
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache}. The returned instance is designed to be used during execution of a
+	 * request thread, so it is not guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>  The type of the cache's keys.
+	 * @param  <V>  The type of the cache's values.
+	 */
+	public abstract <K, V> Cache<K, V> getCache();
+
+	/**
+	 * Returns a new instance of {@link Cache}. The returned instance is designed to be used during execution of a
+	 * request thread, so it is not guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 */
+	public abstract <K, V> Cache<K, V> getCache(int initialCapacity);
+
+	/**
+	 * Returns a new instance of {@link Cache}. The returned instance is designed to be used during execution of a
+	 * request thread, so it is not guaranteed to be {@link java.io.Serializable}. The returned cache will avoid
+	 * exceeding the maximum cache capacity. The algorithm used to ensure that the cache size does not exceed the max
+	 * capacity (for example least-recently-used) may cause significant performance differences between the returned
+	 * cache and a cache returned from {@link #getCache()} or {@link #getCache(int)}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 * @param  maxCapacity      The maximum capacity of the cache.
+	 */
+	public abstract <K, V> Cache<K, V> getCache(int initialCapacity, int maxCapacity);
```

Comment:
@stiemannkj1 Recommend that this method be named `getLRUCache` for the same reasons above.